### PR TITLE
Collect core dumps in more tests

### PR DESCRIPTION
**Pull request by antonio2368**
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### Documentation entry for user-facing changes

- [ ] Documentation is written (mandatory for new features)

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

#### CI Settings (Only check the boxes if you know what you are doing):
- [ ]  Allow: All Required Checks
- [x]  Allow: Stateless tests
- [x]  Allow: Stateful tests
- [ ]  Allow: Integration Tests
- [ ]  Allow: Performance tests
- [ ]  Allow: All Builds
- [ ]  Allow: batch 1, 2 for multi-batch jobs
- [ ]  Allow: batch 3, 4, 5, 6 for multi-batch jobs
---
- [ ]  Exclude: Style check
- [ ]  Exclude: Fast test
- [ ]  Exclude: All with ASAN
- [x]  Exclude: All with TSAN, MSAN, UBSAN, Coverage
- [ ]  Exclude: All with aarch64, release, debug
---
- [ ]  Do not test
- [ ]  Woolen Wolfdog
- [ ]  Upload binaries for special builds
- [ ]  Disable merge-commit
- [ ]  Disable CI cache

### CODE REVIEW

**Review by Algunenano — COMMENTED** [review]
Extra curve ball: Would it be possible to fail the tests if there is a coredump? In the past we've had client/local coredumps and didn't detect it because the tests pass

**Review by Algunenano — APPROVED** [review]

**Comment by robot-ch-test-poll1** [comment]
*This is an automated comment for commit 0821d72f1cef93e437f5b38477f2a2d9d9611f8e with description of existing statuses. It's updated for the latest CI running*

[✅ Click here](https://s3.amazonaws.com/clickhouse-test-reports/66281/0821d72f1cef93e437f5b38477f2a2d9d9611f8e/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Push to Dockerhub</td><td>The check for building and pushing the CI related docker images to docker hub</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/66281/0821d72f1cef93e437f5b38477f2a2d9d9611f8e/push_to_dockerhub_aarch64.html">✅ success</a></td></tr>
<tbody>
</table>
</details>

**Comment by antonio2368** [comment]
@Algunenano that would be logical and I can look into it after we merge this so I can debug jemalloc abort https://github.com/ClickHouse/ClickHouse/issues/66193

**Comment by antonio2368** [comment]
> sysctl: setting key "kernel.core_pattern", ignoring: Read-only file system

**Comment by azat** [comment]
You need `--privileged` docker container for this

**Comment by antonio2368** [comment]
@azat with `--privileged` it seems we don't get `OSReadChars` in `01268_procfs_metrics`

File: tests/queries/0_stateless/01268_procfs_metrics.sh [hunk 1]
```diff
@@ -8,14 +8,6 @@ CURDIR=$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)
...
 . "$CURDIR"/../shell_config.sh
 
-# NOTE: netlink taskstruct interface uses rounding to 1KB [1], so we cannot use ${BASH_SOURCE[0]}
-#
-#   [1]: https://elixir.bootlin.com/linux/v5.18-rc4/source/kernel/tsacct.c#L101
-tmp_path=$(mktemp "$CURDIR/01268_procfs_metrics.XXXXXX")
-trap 'rm -f $tmp_path' EXIT
-truncate -s1025 "$tmp_path"
-
-$CLICKHOUSE_LOCAL --profile-events-delay-ms=-1 --print-profile-events -q "SELECT * FROM file('$tmp_path', 'LineAsString') FORMAT Null" |& grep -m1 -F -o -e OSReadChars
```
Comment (azat) [inline tests/queries/0_stateless/01268_procfs_metrics.sh:18]: The proper fix is to add `--storage_file_read_method=pread`

The reason it started to fail is that the provider switched to netlink and so it does not have noisy `os_read_chars`, like the `procfs` provider has (it reads `/proc`).

**Comment by antonio2368** [comment]
ASAN breaks with `privileged` because of https://github.com/llvm/llvm-project/issues/55944
Setting `sysctl fs.suid_dumpable=1` fixes it but need to verify it's okay

**Pull request merged**